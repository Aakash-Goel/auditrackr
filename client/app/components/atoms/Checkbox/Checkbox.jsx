import React from 'react';
import { object } from 'prop-types';

import withStyles from '@material-ui/core/styles/withStyles';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';

import checkboxStyles from './Checkbox.style';

/**
 * Type checking - Define prop types
 * @private
 */
const propTypes = {
  formControlLabelProps: object,
  checkboxProps: object,
};

/**
 * Define default prop values
 * @private
 */
const defaultProps = {
  formControlLabelProps: {},
  checkboxProps: {},
};

const CustomCheckbox = ({ ...props }) => {
  const { formControlLabelProps, checkboxProps } = props;

  return (
    <FormControlLabel
      {...formControlLabelProps}
      control={<Checkbox color="primary" {...checkboxProps} />}
    />
  );
};

CustomCheckbox.propTypes = propTypes;
CustomCheckbox.defaultProps = defaultProps;

export default withStyles(checkboxStyles)(CustomCheckbox);
