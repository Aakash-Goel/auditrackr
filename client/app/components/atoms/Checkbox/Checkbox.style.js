/**
 * It seems not required. But kept it, if needed in future.
 *
 * If require to modify the styles, then recommended way is to
 * overrides the mui-theme directly.
 *
 * For stylesheet name and other details, check out:
 *  https://material-ui.com/api/checkbox/#css
 *
 */
const checkboxStyles = {};

export default checkboxStyles;
