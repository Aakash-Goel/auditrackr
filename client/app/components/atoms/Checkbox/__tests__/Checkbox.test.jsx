import React from 'react';
import TestRenderer from 'react-test-renderer';

import Checkbox from '../Checkbox';

describe('<Checkbox/>', () => {
  it('renders correctly with empty props', () => {
    const checkboxWrapper = TestRenderer.create(<Checkbox />).toJSON();
    expect(checkboxWrapper).toMatchSnapshot();
  });
});
