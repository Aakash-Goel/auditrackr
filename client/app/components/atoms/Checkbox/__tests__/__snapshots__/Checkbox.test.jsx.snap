// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox/> renders correctly with empty props 1`] = `
<label
  className="MuiFormControlLabel-root-2"
>
  <span
    className="MuiButtonBase-root-24 MuiIconButton-root-18 MuiPrivateSwitchBase-root-14 MuiCheckbox-root-8 MuiCheckbox-colorPrimary-12"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label-23"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-27"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <input
        className="MuiPrivateSwitchBase-input-17"
        data-indeterminate={false}
        onChange={[Function]}
        type="checkbox"
      />
    </span>
    <span
      className="MuiTouchRipple-root-72"
    />
  </span>
  <span
    className="MuiTypography-root-36 MuiTypography-body1-45 MuiFormControlLabel-label-7"
  />
</label>
`;
